jpdias/node-red-contrib-self-healing
# Integrate changes from LDSO T4G2 group

- Add CHANGELOG
- Fix Dockerfile
- style: add Prettier to project, as well as format and format-check scripts
- style: run format on entire project
- ci: Add format checking pipeline job
- test: Add example of unit test for node
- Fix merge conflicts with dev
- style: run format
- ci: Add unit testing pipeline job
- feat: add initial implementation of balancing node
- feat: Clean and expand properties window
- ci: Add Dockerfile for GitHub deploy Docker image
- ci: Add manual deploy job to open PR on GitHub
- refactor: Join operators and their descriptions into one object
- style: Enforce linter rules
- feat: add dynamic number of outputs
- style: Make help section more readable
- refactor: Make html code more uniform
- style: Change text type and script order to match the ones in Node-RED docs
- style: Format threshold-check code
- ci: Add mocha globals to ESLint env
- refactor: split methods into smaller ones
- style: allow unused vars on ESLint as long as prefixed with underscore
- fix: Fix files that were triggering linting errors
- ci: Enable linting on test stage of pipeline
- refactor: update node help section
- style: change text type and script order to match Node-RED docs
- refactor: change variable declaration keywords
- style: Change text type and script order to match the ones in Node-RED docs
- style: html formatting
- feat: timing check node reimplemented
- feat: Add Round Robin algorithm
- style: Format replication-voter code
- style: Prettier format timing check js and html code
- refactor: Change var to let
- style: Format replication-voter code
- fix: Errors introduced by previous var keywords
- test: Add two base tests
- test: Add tests for all unary and binary operations
- style: formatting according to prettier
- docs: documentated timing check node's functionality in its html
- feat: reimplemented http connection heartbeat
- fix: error messages displaying correctly
- refactor: split methods into smaller ones
- fix: Regex not working
- fix: Error with prev and mean values
- test: Add remaining tests
- style: Format code
- feat: save/load strategy checkbox state
- refactor: clean existing (min/max) strategies
- fix: add return to mode method
- feat: show/hide properties of enabled/disabled strategies
- feat: Implement 'stuck at same reading' strategy
- feat: Add Weighted Round Robin algorithm
- docs(replication-voter): add node documentation
- docs: corrected node's documentation format
- style: ran prettier format in timing check's html
- docs: Add description for the node
- feat: mqtt protocol and basis for testing
- fix: fix of variable names and html content
- style: format changes
- feat: fill out node documentation
- refactor: changes to message creation
- test: Add tests to check functionality of node
- feat: show target flows available for target host and port
- docs: remove documented type of flow target property
- refactor: remove duplicate code and update html description
- fix: Fix lexical declaration in case blocks
- style: format code with prettier
- Refactor: Remove unnecessary comments
- test: Add support for code coverage analysis
- test: Add support for mutation testing
- style: Format stryker config file
- fix: more correct handling of invalid input
- test: Add tests to check node functionality
- style: Format code
- refactor: Move BoundedStack class into utils folder
- feat: Clean and expand properties window
- feat: save/load strategy checkbox state
- refactor: clean existing (min/max) strategies
- feat: show/hide properties of enabled/disabled strategies
- feat: Implement 'stuck at same reading' strategy
- feat: fill out node documentation
- fix: more correct handling of invalid input
- test: Add tests to check node functionality
- style: Format code
- refactor: Move BoundedStack class into utils folder
- ci: Add support for sast
- docs: add comments and html node information
- debug: Add debug to CI
- fix: Wrong jobname in CI
- fix: Remove template and add njsscan manualy
- test: Add tests to check functionality of node
- ci: SAST only runs on master
- style: Change text type and script order to match the ones in Node-RED docs
- style: Format replication-voter code
- style: Format replication-voter code
- refactor: split methods into smaller ones
- fix: add return to mode method
- docs(replication-voter): add node documentation
- test: Add tests to check functionality of node
- test: Add missing test to node balancing
- fix: remove unused variables
- fix: format code to fix lint errors
- test: testing status messages
- test: add a simple test
- refactor: removed unnecessary comment
- test: implemented tests for timing node check
- refactor: Improve code
- test: add edge case tests
- style: run npm format
- fix: properly shut down node on close event
- style: prettier format
- refactor: refactor constant variables from let to const declaration
- refactor: replace let with const keyword
- refactor: removed useless try catch statement on timing-check_spec.js
- style: prettier format run
- build: add package-lock.json to Dockerfile
- style(eslint): allow unused variables with _ prefix
- refactor(kalman): Refactor code
- docs(kalman): Overhaul node documentation
- test(kalman): add 2 unit tests for node
- build(kalman): add kalman node to Dockerfile
- test(kalman): test multiple single-number payloads
- docs: updated sprint 0 and sprint 1 changelog
- style: format code with prettier
- docs: updated CHANGELOG.md
- test: add test for default behaviour and allByOrder
- fix: remove code to fix lint errors
- refactor: remove unecessary code and update node info
- fix(ci): fix github deploy script
